### PR TITLE
[REEF-406]: Give user an option to configure the TCPPortProvider in B…

### PR DESCRIPTION
**Pull request by dkm2110**
[REEF-406]: Give user an option to configure the TCPPortProvider in BroadcastAndReduceClient and PipelineBroadcastAndReduceClient

This addressed the issue by
	* Allowing user to specify the starting port number and range via command line.
	* Binding this information in the driver configuration.

JIRA: [REEF-406](https://issues.apache.org/jira/browse/REEF-406)

### CODE REVIEW

**Comment by markusweimer** [comment]
This looks good, I'll merge

File: lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs [hunk 1]
```diff
@@ -52,17 +53,21 @@ public class PipelinedBroadcastReduceDriver : IStartHandler, IObserver<IEvaluato
...
+        private readonly int _arraySize;
 
         private readonly IGroupCommDriver _groupCommDriver;
         private readonly ICommunicationGroupDriver _commGroup;
         private readonly TaskStarter _groupCommTaskStarter;
+        private readonly IConfiguration _tcpPortProviderConfig;
 
         [Inject]
         public PipelinedBroadcastReduceDriver(
             [Parameter(typeof (GroupTestConfig.NumEvaluators))] int numEvaluators,
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs]: remove space

File: lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs [hunk 2]
```diff
@@ -52,17 +53,21 @@ public class PipelinedBroadcastReduceDriver : IStartHandler, IObserver<IEvaluato
...
-            [Parameter(typeof(GroupTestConfig.NumIterations))] int numIterations,
-            [Parameter(typeof(GroupTestConfig.ChunkSize))] int chunkSize,
+            [Parameter(typeof (GroupTestConfig.NumIterations))] int numIterations,
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs]: remove space after typeof

**Comment by jwang98052** [comment]
I complete my pass. Looks good.

File: lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs (hunk 2 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs]: I'll do that as part of the merge

Comment (dkm2110) [inline lang/cs/Org.Apache.REEF.Network.Examples/GroupCommunication/PipelineBroadcastReduceDriverAndTasks/PipelinedBroadcastReduceDriver.cs]: Thanks

**Pull request closed without merging**